src/multiplicity_vectors.lean

```diff
@@ -0,0 +1,645 @@
+/-
+Copyright (c) 2020 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Aaron Anderson.
+-/
+
+import data.finsupp
+import data.pnat.factors
+import order.conditionally_complete_lattice
+import order.lattice
+import order.order_iso
+import tactic.pi_instances
+
+/-!
+# Factor Finsupps
+
+A new version of prime factorisation and some facts about lattice isomorphisms
+
+## Notations
+
+## Implementation Notes
+factor_finsupp is similar to pnat.factor_multiset, but a finsupp of exponents
+Most of the sorries will be solved by proving some facts about lattice isomorphisms.
+
+## References
+
+-/
+
+set_option old_structure_cmd true
+
+open_locale classical
+open_locale big_operators
+
+
+noncomputable theory
+
+section basics
+
+def pnat.coprime (m n : ℕ+) : Prop := m.gcd n = 1
+
+def mult (n : ℕ+) (p : nat.primes) : ℕ := (pnat.factor_multiset n).count p
+
+def prime_finsupp : Type := nat.primes →₀ ℕ
```

Comment:
I'm not sure that this should be a new definition. I would just use it as is.

Reply:
Which are you suggesting, that I get red of mult or prime_finsupp? I realized I never ended up using mult, and deleted it, but prime_finsupp I'm using (for now) in analogy to prime_multiset and the warning on the main finsupp page that you should probably wrap it in another structure. I'm still open to getting rid of it.

Reply:
Aah, I didn't know about that warning. However, in this case I actually think you should ignore the warning. I guess that's more directed at people who want to use the algebra structure (monoid algebra, polynomials). But you aren't really adding extra structure on top, and moreover, you are fixing both arguments to `finsupp` (namely `nat.primes` and `nat`). So I don't really see the need for a new definition. Every new definition needs supporting lemmas that connect it to existing definitions... I think we can save on those here.
I've seen that you are using this definition a lot. But I think you can just copy-paste-replace `nat.primes →₀ ℕ` wherever you use it (maybe with parens). My personal impression is that the result will be more readable.

Reply:
@jcommelin do you mind adding comments along the above lines to the 1finsupp` module doc? I also found the text there a bit confusing.

Reply:
@bryangingechen I'll try to make a PR tomorrow.

Reply:
I can do that, but renaming all of the functions in the prime_finsupp namespace might get a little weird